src/__tests__/useRaf.test.ts

```diff
@@ -0,0 +1,153 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import { replaceRaf } from 'raf-stub';
+import useRaf from '../useRaf';
+
+/**
+ * New requestAnimationFrame after being replaced with raf-stub for testing purposes.
+ */
+interface RequestAnimationFrame {
```

Comment:
On this test I'm completely mocking RAF and simulating only the minimum behavior I need.